internal/rbd/nodeserver.go

```diff
@@ -1227,26 +1234,70 @@ func (ns *NodeServer) NodeGetVolumeStats(
 	ctx context.Context,
 	req *csi.NodeGetVolumeStatsRequest,
 ) (*csi.NodeGetVolumeStatsResponse, error) {
-	var err error
-	targetPath := req.GetVolumePath()
-	if targetPath == "" {
-		err = fmt.Errorf("targetpath %v is empty", targetPath)
+	var (
+		imgInfo rbdImageMetadataStash
+		err     error
+	)
+	volumePath := req.GetStagingTargetPath()
+	if volumePath == "" {
+		if volumePath = req.GetVolumePath(); volumePath == "" {
+			return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
+		}
+	} else {
+		volumePath = volumePath + "/" + req.GetVolumeId()
+		imgInfo, err = lookupRBDImageMetadataStash(volumePath)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to find image metadata at stagingPath %q: %v", volumePath, err)
 
-		return nil, status.Error(codes.InvalidArgument, err.Error())
+			return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
why this is changed to `NotFound` error code?

Reply:
Not found is not suitable for all errors. if error is `ErrMissingStash` you can consider it as not for. keeping Internal error is also fine.

Reply:
Done now.